From 94412eef0fcaf1b5ac5d0367e25b7ce54b33845c Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Tue, 31 May 2022 14:01:45 +0200
Subject: [PATCH 062/176] fix(linstor-manager): change linstor satellite start
 behavior

Ensure we don't have an invalid cache used by a satellite:
- We found an issue with a new added disk which used a volume group name
  formerly involved by another disk. To avoid this kind of problem, we
  always restart the satellite.

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/linstor-manager | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/linstor-manager b/drivers/linstor-manager
index 7893ebc..c6d622f 100755
--- a/drivers/linstor-manager
+++ b/drivers/linstor-manager
@@ -61,7 +61,16 @@ def update_all_ports(open):
 
 
 def update_linstor_satellite_service(start):
-    util.enable_and_start_service('linstor-satellite', start)
+    service = 'linstor-satellite'
+
+    # Stop services in all cases first.
+    # Ensure we don't have an invalid cache used by a satellite.
+    # (We found an issue with a new added disk which used a volume group name
+    # formerly involved by another disk. To avoid this kind of problem, we
+    # always restart the satellite.)
+    util.enable_and_start_service(service, False)
+    if start:
+        util.enable_and_start_service(service, True)
 
 
 def update_minidrbdcluster_service(start):
-- 
2.45.2

